examples/lodash/should_migrate.js

```diff
@@ -0,0 +1,48 @@
+const semver = require('semver');
+const path = require('path');
+const fs = require('fs');
+
+const pkg = require(path.join(process.cwd(), 'package.json'));
+
+const version3deps = [];
+const version4deps = [];
+const dependencies = pkg['dependencies'];
+if (dependencies) {
+  Object.keys(dependencies).forEach(pkgName => {
+    if (pkgName === 'lodash' || pkgName.startsWith('lodash.')) {
+      const version = dependencies[pkgName];
+      if (semver.ltr('3.999.999', version)) {
+        // We're on v4 already
+        version4deps.push(pkgName);
+      } else if (semver.ltr('2.999.999', version)) {
+        version3deps.push(pkgName);
+      } else {
+        throw new Error(`Unknown package version: ${pkgName} ${version}`);
+      }
+    }
+  })
+}
+
+if (!version3deps.length && !version4deps.length) {
+  console.error('This repo does not depend on Lodash');
+  process.exit(1);
```

Comment:
Aside: I wonder if there's a reasonable way for shepherd to distinguish between a real error in this script and just exiting 1 to indicate *should not migrate*.